src/ui.cpp

```diff
@@ -127,6 +127,71 @@ uimenu::uimenu(bool cancelable, int startx, int width, int starty, std::string t
     query();
 }
 
+uilist::uilist()
+{
+    init();
+}
+
+uilist::uilist( std::string hotkeys_override )
+{
+    init();
+    if( !hotkeys_override.empty() ) {
+        hotkeys = std::move( hotkeys_override );
```

Comment:
`std::move` doesn't help much here: a copy has already been made since you don't accept the argument by an rvalue reference.

Reply:
My thought was, if I accept a `const std::string &`, then when someone calls the constructor with a `const char *`, first a `std::string` will be constructed as the argument, then the constructor copies the argument to the member value, which is twice copying the actual text. If I accept a `std::string`, only one copy is made, and then I `std::move` the value to the member value. I did not use `std::string &&` because it would prevent lvalue references from being passed. I could of course create more constructors accepting C-style string, `const std::string &`, and `std::string &&` parameters, but didn't think it would worth the typing.

Reply:
Please eschew std::move, it makes code needlessly fragile for no measurable benefit (in most application contexts).

Reply:
I can see the argument that the benefit is small, but I don't see why it would make code fragile. From my understanding `std::move` casts an lvalue reference to an rvalue reference, which marks that the object's content is safe for reuse. And in this case the parameter's value is never used after the move, so it'll be perfectly safe.

Reply:
If you routinely use std::move(), you will create a situation where you or someone else accidentally attempts to reuse moved variables, and due to the semantics of std::move these uses are silent and the failures are a pain to troubleshoot.  It's happened a number of times in this project already, and the benefit is so small-to-nonexistent that it's not worth it to mess with as a language feature unless we have strong evidence it's necessary.

Reply:
I've changed all constructors to take const lvalue references instead.